src/documents/PromisesAPI.md

```diff
@@ -0,0 +1,236 @@
+
+Coerces any value to a promise. If it is already a promise, it is returned unchanged. Otherwise, `toPromise` constructs a fresh promise resolving to the supplied value.
+
+
+### fluid.promise.follow(source, target)
+
+* `source {Promise}` A promise which is to be followed in its resolution.
+* `target {Promise}` A promise which will follow the `source` in its resolution. `target` will receive a call to `then` causing it to resolve when `source` resolves,
+and reject when `source` rejects. 
+
+Chains the resolution methods of one promise (target) so that they follow those of another (source).
+That is, whenever source resolves, target will resolve, or when source rejects, target will reject, with the
+same payloads in each case.
+
+### fluid.promise.map(source, func)
+
+Returns a promise whose resolved value is mapped from the source promise or value by the supplied function. If the input value is not a promise, it will
+be converted first to a promise via `fluid.toPromise()`. If the input promise rejects, its rejection reason will be propagated unmapped. 
+
+* `source {Object|Promise}` An object or promise whose value is to be mapped (if an object, will be converted first to a promise via `fluid.toPromise()`)
+* `func {Function: ({Any})  →  Any}` A function which will map the resolved promise value
+* Returns: `{Promise}` A promise for the resolved mapped value.
+ 
+## Promise algorithms
+
+The only currently implemented promise algorithms are based around a core skeleton operating an array of promises in a linear sequence. These 
+are responsive to an additional element of our promises API, the  [`fluid.accumulateRejectionReason`](#promise.accumulaterejectionreason-error-) "inverse API" described below.
+
+### fluid.promise.sequence(sources[, options])
+
+* `sources {Array of {Any|Promise|Function:(options {Object}) →  Any|Promise}}` An array of sources of values or promises which will be evaluated in sequence.
+* `options {Object}` [optional] A structure of options which will be supplied to function members of `sources`.  
+
+Accepts an array of values, promises, functions returning values or functions returning promises and evaluates them in sequence. Note that a standard name for a "function returning a promise" 
+is a *task* - this implementation can be directly compared to [sequence](https://github.com/cujojs/when/blob/master/docs/api.md#whensequence) in the [when.js library](https://github.com/cujojs/when).
+
+In the case that the source element is a function returning a promise (a task), `fluid.promise.sequence` will ensure that
+at most one of these in "in flight" at a time - that is, the succeeding function will not be invoked
+until the promise at the preceding position has resolved.   
+
+### fluid.promise.fireTransformEvent(event, payload[, options])
+
+* `event {[Event](InfusionEventSystem.md)}` A "pseudoevent" whose listeners are to be treated as sucessive (asynchronous) stages in the process of transforming a payload.
```

Comment:
This link to InfusionEvenSystem.md isn't rendering as a link (I think because it's inside the code styling)